pkg/process/metadata/parser/java/spring.go

```diff
@@ -0,0 +1,205 @@
+	if !(filepath.IsAbs(path) || path[0] == os.PathSeparator) && len(cwd) > 0 {
+		return filepath.Join(cwd, path)
+	}
+	return path
+}
+
+// newSpringBootArchiveSourceFromReader return a PropertyGetter combined source with properties sources from the jar application.
+func newSpringBootArchiveSourceFromReader(reader *zip.Reader, patternMap map[string][]string) map[string]*props.Combined {
+	ret := make(map[string]*props.Combined)
+	// probably we need something more efficient
+	matcher := antpath.New()
+	for _, f := range reader.File {
+		name := f.Name
+		// the generalized approach implies visiting also jar in BOOT-INF/lib but here we skip it
+		// to minimize the scanning time given that the general habit is to package config
+		// directly into the application and not in a lib embedded into the app
+		if strings.HasPrefix(name, "BOOT-INF/classes/") {
+			for profile, patterns := range patternMap {
+				for _, pattern := range patterns {
+					if matcher.Match(pattern, name) {
+						rc, err := f.Open()
+						if err != nil {
+							log.Trace("Unable to open", name, ":", err)
+						} else {
+							source, err := func() (*props.PropertyGetter, error) {
+								defer rc.Close()
+								source, err := newPropertySourceFromStream(rc, f.Name, f.UncompressedSize64)
+								return source, err
+							}()
+							if err != nil {
+								log.Trace("Error creating property source", err)
+							} else {
+								log.Trace("Loaded", f.Name)
+								val, ok := ret[profile]
+								if !ok {
+									val = &props.Combined{Sources: []props.PropertyGetter{}}
+									ret[profile] = val
+								}
+								val.Sources = append(val.Sources, *source)
+							}
+						}
+						break
+					}
+				}
+			}
+		}
+	}
+	return ret
+}
+
+// GetSpringBootAppName tries to autodetect the name of a spring boot application given its working dir,
+// the jar path and the application arguments
+func GetSpringBootAppName(cwd string, jarname string, args []string) (string, error) {
+	absName := abs(jarname, cwd)
+	log.Tracef("open jar archive %s", absName)
+	archive, err := zip.OpenReader(jarname)
+	if err != nil {
+		return "", err
+	}
+	defer archive.Close()
+	ok, err := IsSpringBootArchive(&archive.Reader)
+	if !ok || err != nil {
+		log.Trace("Not a spring boot archive.")
+		return "", nil
+	}
+
+	combined := &props.Combined{Sources: []props.PropertyGetter{
+		newArgumentSource(args, "--"),
+		newArgumentSource(args, "-D"),
+		// TODO: add the env variable of the process being introspected
+		// I did not found in the agent packages something to do it cross os.
+		// On linux one trivial way is to look in the proc fs
+		//&props.Environment{Normalize: true},
+	}}
+
+	// resolved properties referring to other properties just in case...
+	conf := &props.Configuration{Props: props.NewExpander(combined)}
+	// Looking in the environment (sysprops, arguments) first
+	appname, ok := conf.Get(appnamePropName)
+	if !ok {
```

Comment:
The nesting here makes the code harder to read
```suggestion
	if ok {
		return ...
	}
	....
```

Reply:
changed accordingly